### DISCUSSION TITLE
How did dev of this integration start? Where do you source documentation for BLE protocol and etc?

### DISCUSSION THREAD

**Discussion by DjordjeMandic** [home-assistant-HomeWhiz/home-assistant-HomeWhiz#209] [General]
Question same as in title

**Comment by rowysock** [comment]
Hi, I started working on that integration about 2 years ago, so my memory is a little bit blurry, but my main sources of knowledge were:
* BLE packet sniffing and Wireshark analysis
* Decompiled original app source code analysis

Initially, I thought that I would be able to deduct a lot of stuff from just the BLE packets themself, but they turned out to be far too complex for me to decipher, so in the end most of the knowledge came from decompiled source code and BLE recordings served more as a debugging tool.
